OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -442,20 +441,18 @@ public override bool Tick(Actor self)
 			{
 				var mobile = Move.mobile;
 
-				// Having non-zero progress in the first tick means that this MovePart is following on from
-				// a previous MovePart that has just completed during the same tick. In this case, we want to
-				// apply the carried over progress but not evaluate a full new step until the next tick.
-				if (!firstTick || progress == 0)
+				// Only move by a full speed step if we didn't already move this tick.
+				// If we did, we limit the move to any carried-over leftover progress.
+				if (Move.lastMovePartCompletedTick < self.World.WorldTick)
 					progress += mobile.MovementSpeedForCell(self, mobile.ToCell);
 
-				firstTick = false;
-
 				if (progress >= Distance)
 				{
 					mobile.SetCenterPosition(self, To);
 					mobile.Facing = ToFacing;
 
 					Queue(OnComplete(self, mobile, Move));
+					Move.lastMovePartCompletedTick = self.World.WorldTick;
```

Comment:
Minor nit: It would be more robust to set this before calling `OnComplete`, as this makes sure that any future code changes that might read `Move.lastMovePartCompletedTick` from the `MovePart` ctor see the same value.